File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static List<List<Integer>> graph;
    static boolean[] visited;

    private static void dfs(int node) {
        System.out.print(node + " ");
        visited[node] = true;

        for (int neighbor : graph.get(node)) {
            if (!visited[neighbor]) dfs(neighbor);
        }
    }

    private static void bfs(int node) {
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(node);
        visited[node] = true;

        while (!queue.isEmpty()) {
            int cur = queue.poll();
            System.out.print(cur + " ");

            for (int neighbor : graph.get(cur)) {
                if (!visited[neighbor]) {
                    visited[neighbor] = true;
                    queue.offer(neighbor);
                }
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int start = Integer.parseInt(st.nextToken());

        graph = new ArrayList<>();
        for (int i = 0; i <= n; i++) {
            graph.add(new ArrayList<>());
        }

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int u = Integer.parseInt(st.nextToken());
            int v = Integer.parseInt(st.nextToken());
            graph.get(u).add(v);
            graph.get(v).add(u);
        }

        for (int i = 1; i <= n; i++) {
            Collections.sort(graph.get(i));
        }

        visited = new boolean[n + 1];
        dfs(start);
        System.out.println();

        visited = new boolean[n + 1];
        bfs(start);
        System.out.println();
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.LinkedList
import java.util.Queue

lateinit var graph: MutableList<MutableList<Int>>
lateinit var visited: BooleanArray

fun dfs(node: Int) {
    print("$node ")
    visited[node] = true

    for (neighbor in graph[node]) {
        if (!visited[neighbor]) dfs(neighbor)
    }
}

fun bfs(node: Int) {
    val queue: Queue<Int> = LinkedList()
    queue.offer(node)
    visited[node] = true

    while (queue.isNotEmpty()) {
        val cur = queue.poll()
        print("$cur ")

        for (neighbor in graph[cur]) {
            if (!visited[neighbor]) {
                visited[neighbor] = true
                queue.offer(neighbor)
            }
        }
    }
}

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val (n, m, start) = br.readLine().split(' ').map { it.toInt() }
    graph = MutableList(n + 1) { mutableListOf() }

    repeat(m) {
        val (u, v) = br.readLine().split(' ').map { it.toInt() }
        graph[u].add(v)
        graph[v].add(u)
    }

    for (i in 1..n) {
        graph[i].sort()
    }

    visited = BooleanArray(n + 1)
    dfs(start)
    println()

    visited = BooleanArray(n + 1)
    bfs(start)
    println()
}